File: array_problems/reverseArray/reverseArray.py
class ArrayProblems:
    def __init__(self, arr):
        self.arr = arr.copy()

    def ReverseArrA(self):
        response_arr = []
        for i in range(len(self.arr) - 1, -1, -1):
            response_arr.append(self.arr[i])
        return response_arr

    def ReverseArrB(self):
        left = 0
        right = len(self.arr) - 1
        while left < right:
            self.arr[left], self.arr[right] = self.arr[right], self.arr[left]
            left += 1
            right -= 1
        return self.arr

arr_problems = ArrayProblems([0, 43, 3, 2, 3, 4, 6])
print(arr_problems.ReverseArrA())
print(arr_problems.ReverseArrB())

File: array_problems/reverseArray/reverseArray.cs
using System;

public class ArrayProblem
{
    private int[] arr;

    public ArrayProblem(int[] arr)
    {
        this.arr = arr;
    }

    public void SetArr(int[] arr)
    {
        this.arr = arr;
    }

    public int[] ReverseArrA()
    {
        int[] responseArr = new int[arr.Length];
        for (int i = arr.Length - 1, j = 0; i >= 0; i--, j++)
        {
            responseArr[j] = arr[i];
        }
        return responseArr;
    }

    public int[] ReverseArrB()
    {
        int start = 0;
        int end = arr.Length - 1;
        while (start < end)
        {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
        return arr;
    }

    public static void Main(string[] args)
    {
        ArrayProblem arrProblems = new ArrayProblem(new int[0]);
        arrProblems.SetArr(new int[] { 0, 43, 3, 2, 3, 4 });
        Console.WriteLine("Solution A: " + string.Join(", ", arrProblems.ReverseArrA()));
        Console.WriteLine("Solution B: " + string.Join(", ", arrProblems.ReverseArrB()));
    }
}